### DISCUSSION TITLE
Missing matplotlib samples

### DISCUSSION THREAD

**Discussion by frozax** [pthom/imgui_bundle#193] [Q&A] [answered]
Hello,
I'm using matplotlib in imgui-bundle and the provided sample is a great starting point. However, I think it lacks some features that are probably there but couldn't find doc/samples about it. For example, is it possible to shrink the figure automatically to the available space in the window, detecting clicks and drags on the figures, adding zooming and panning...
Or maybe using matplotlib is not really a good idea and I should focus on implot?
Thanks

**Comment by pthom** [comment, marked as answer]
Hello,

Given your use case, yes I would say that you should focus on implot, because handling zooming and panning would be difficult, especially given that the matplotlib API is not very adapted to an immediate gui.

**Reply by frozax** [reply]
Thanks for your quick reply. I'll have a look at the possibilities of implot, then! I planned to learn it anyway ;)
Just to explain why I focused on matplotlib initially: I'm using libraries with deep integration with matplotlib. For example librosa that can display sound wave and spectrograms natively.
